src/org/jetbrains/plugins/scala/lang/psi/impl/toplevel/typedef/StalactiteInjector.scala

```diff
@@ -0,0 +1,57 @@
+package org.jetbrains.plugins.scala.lang.psi.impl.toplevel.typedef
+
+import org.jetbrains.plugins.scala.lang.psi.api.toplevel.typedef._
+import org.jetbrains.plugins.scala.lang.psi.api.expr.ScAnnotation
+
+/**
+ * Support for https://github.com/fommil/stalactite
+ *
+ * Chooses to skip typeclass derivation (and requirements) in order to
+ * provide a much faster editing experience. Users may experience
+ * deriving failures when they run the real compiler.
+ *
+ * @author Sam Halliday
+ * @since  24/08/2017
+ */
+class StalactiteInjector extends SyntheticMembersInjector {
+  // fast check
+  def hasStalactite(source: ScTypeDefinition): Boolean =
+    source.findAnnotationNoAliases("stalactite.deriving") != null
+
+  // slower, more correct
+  def stalactite(source: ScTypeDefinition): ScAnnotation = {
+    val candidates = source.annotations("stalactite.deriving")
+    if (candidates.size != 1)
+      throw new IllegalArgumentException("expected a single @deriving annotation")
```

Comment:
What should happen in the following scenario (two annotations)?
```
import stalactite._

@deriving(Encoder, Decoder)
@deriving(Encoder, Decoder)
case class Bar(s: String, b: Boolean)
```

I'm not sure what the right behavior should be but throwing an error doesn't seem like the right one. 
If the code above would cause a compile error, it would be better to report it but it might be too hard for this PR.
If the code above would not cause a compile error, we should try to mimic stalactite's behavior as much as we can.

Reply:
That's not allowed. One `@deriving` only... although I'm not 100% sure what macro-paradise does when multiple macro annotations are added to a class.

Reply:
So I think the best thing would be to highlight it as an error but I don't think there's an easy way to do this here. For now, I just would not throw an error

Reply:
I'll just take the first one, how about that?